Login landing

kjrcda opened the pull request:
I am not sure how to write tests for this thing. It doesn't do anything except accept a name through the parameters and display a home page upon logging in.

kjrcda committed:
Landing page

- page after login is setup
- has hooks to match/edit info but doesnt do anything yet
- shows name after login
- some CSS stuff for centered divs and circular buttons

kjrcda committed:
Landing page

- page after login is setup
- has hooks to match/edit info but doesnt do anything yet
- shows name after login
- some CSS stuff for centered divs and circular buttons

kjrcda committed:
Landing page

- page after login is setup
- has hooks to match/edit info but doesnt do anything yet
- shows name after login
- some CSS stuff for centered divs and circular buttons

kjrcda committed:
Landing page

- page after login is setup
- has hooks to match/edit info but doesnt do anything yet
- shows name after login
- some CSS stuff for centered divs and circular buttons

kjrcda committed:
Landing page

- page after login is setup
- has hooks to match/edit info but doesnt do anything yet
- shows name after login
- some CSS stuff for centered divs and circular buttons

kjrcda committed:
Landing page

- page after login is setup
- has hooks to match/edit info but doesnt do anything yet
- shows name after login
- some CSS stuff for centered divs and circular buttons

kjrcda committed:
Landing page

- page after login is setup
- has hooks to match/edit info but doesnt do anything yet
- shows name after login
- some CSS stuff for centered divs and circular buttons

kyleleung75 committed:
Merge pull request #35 from paymog/persist-user

Persist the logged in user. WOOOOOOO another merge!

kjrcda committed:
resolving conflictions

kjrcda committed:
Merge pull request #36 from paymog/development

Weekly Merge Mar 6

paymog committed:
Added a sandwich favicon

paymog commented on webapp/public/javascripts/controllers.js:
```diff
@@ -15,12 +15,13 @@ lunchrControllers.controller('MainPageController', ['$scope', '$http', '$state',
-            $http.post('/api/users/authenticate', {email: $scope.email, password: $scope.password}).
-                success(function (data, status, headers, config) {
+            $http.post('/api/users/authenticate', {email: $scope.email, password: $scope.password})
+                .success(function (data, status, headers, config) {
```
is there a reason you moved the `.`?

kjrcda replied:
everywhere else it seems the . goes inline with the function, not on the line before. since I was in the file i just moved it.

paymog commented on webapp/public/javascripts/controllers.js:
```diff
@@ -79,4 +80,18 @@ lunchrControllers.controller('UserMatchingController', ['$state', 'socket', 'aut
+    function ($scope, $state, $stateParams) {
+        $scope.userName = $stateParams.name;
+        $scope.match = function () {
+            //$state.go('match');
```
remove commented code

paymog commented on webapp/views/partials/home.jade:
```diff
@@ -0,0 +1,8 @@
+div(class='container')
+    h1 {{userName}}
+
+    div(class='div-center')
```
does `text-center` fit your needs here? It's used in `main.jade` and it seems to keep everything centered. If so, you should also remove the css for `div-center` in the css file.

kjrcda replied:
text-center only centers text is seems. This div-center I made centers everything inside the div, or at least the button I made.

paymog commented on webapp/views/partials/home.jade:
```diff
@@ -0,0 +1,8 @@
+    h1 {{userName}}
+
+    div(class='div-center')
+        button(type='button', class='btn-circle', style='margin-bottom: 30px; margin-top: 50px;' ng-click='match()') Find a Match
```
I think a general rule of thumb is to not put style information in HTML. It might be good to give the button a name (tag?) and providing styling for that name in the css file.

paymog replied:
not name or tag, id. See http://www.w3schools.com/cssref/sel_id.asp

paymog replied:
missing comma between `style` and `ng-click`

paymog commented on webapp/views/partials/home.jade:
```diff
@@ -0,0 +1,8 @@
+    div(class='div-center')
+        button(type='button', class='btn-circle', style='margin-bottom: 30px; margin-top: 50px;' ng-click='match()') Find a Match
+
+    div(class='div-center' style='width: 12em;')
```
same as above regarding style

paymog commented:
The tests are failing. Specifically, `$scope.login` for `MainPageController`

kjrcda commented:
I am not sure what to do about the main page controller test.  when I do a scope.login, the user gets returned as a result normally. I am not sure how to mock this in the test though. I then pass first and last name to the home page (currently, this will change with the authService).

paymog commented:
Look at line 46 in `main_page_controller_test` and change how the mocked post responds.

kjrcda committed:
tests pass + css cleanup

- cleaned up CSS
- Updated tests to pass

kjrcda committed:
tests pass + css cleanup

- cleaned up CSS
- Updated tests to pass

kjrcda committed:
tests pass + css cleanup

- cleaned up CSS
- Updated tests to pass

kjrcda committed:
delete comments

kjrcda committed:
delete comments

kjrcda committed:
delete comments

kjrcda committed:
delete comments

paymog commented on webapp/ng_tests/controllers/main_page_controller_test.js:
```diff
@@ -37,14 +37,14 @@ describe('MainPageController', function () {
 
-            $httpBackend.expectPOST('/api/users/authenticate').respond(200, '');
-            $state.expectTransitionTo('users');
+            $httpBackend.expectPOST('/api/users/authenticate').respond(200, [{email: 'user', password: 'password', firstname: 'first', lastname: 'last'}]);
```
Why pass email and password if they're not being used? Should a separate test be made outlining what's expecting as a response?

paymog commented on webapp/public/javascripts/controllers.js:
```diff
@@ -79,4 +80,17 @@ lunchrControllers.controller('UserMatchingController', ['$state', 'socket', 'aut
+
+lunchrControllers.controller('HomePageController', ['$scope', '$state', '$stateParams',
+    function ($scope, $state, $stateParams) {
+        $scope.userName = $stateParams.name;
```
You should have a test for this line.
Also, why rely on `$stateParams` and not `authService`?

paymog replied:
Actually, nevermind about the `authService`

kyleleung75 committed:
Merge pull request #39 from paymog/favicon

Added a sandwich favicon, booyah it works

kyleleung75 committed:
Merge pull request #37 from paymog/persist-user

There wasn't actual persistence, this is fixed, BOOYAH it works.

kjrcda committed:
Quick test cleanup

Got rid of extra parameters in main controller test. Made a test file
for Home Page Controller

kjrcda committed:
Quick test cleanup

Got rid of extra parameters in main controller test. Made a test file
for Home Page Controller

kjrcda committed:
Quick test cleanup

Got rid of extra parameters in main controller test. Made a test file
for Home Page Controller

kjrcda committed:
fixed conflict in controller

BrandonWade committed:
Merge pull request #41 from paymog/persist-user

Fix tests

kjrcda committed:
landing done

- No longer uses params, uses authService now
- Modified /api/users get call to look for single user
- Found redundant HomePageController in controller file so i deleted it

kjrcda committed:
landing done

- No longer uses params, uses authService now
- Modified /api/users get call to look for single user
- Found redundant HomePageController in controller file so i deleted it

kjrcda committed:
GET in sync

Just made the GET request follow same format as all the other $http.get
calls.

kjrcda committed:
GET in sync

Just made the GET request follow same format as all the other $http.get
calls.

kjrcda committed:
Merge branch 'login-landing' of https://github.com/paymog/Lunchr into login-landing

kjrcda committed:
Updated test

Test actually does stuff now

paymog commented on webapp/ng_tests/controllers/home_page_controller_test.js:
```diff
@@ -0,0 +1,49 @@
+    describe('on initialization', function() {
+        it('sets $scope.name based on call to db using authService', function() {
+
+            $httpBackend.expectGET('api/users?email=' + DEFAULT_EMAIL).respond(200, [{firstname: DEFAULT_FIRSTNAME, lastname: DEFAULT_LASTNAME}]);
```
Can't you do something like: `expectGet('api/users', {email: DEFAULT_EMAIL)`?

kjrcda replied:
I tried, that doesn't work. The test expects 'api/users' but we need to expect 'api/users?email.....'

paymog commented:
Another way to architect this solution is to:
1) Whenever a use is signed in (either through log in or register) have the server return the entire user object
2) Store this user object inside the auth service
3) User what's stored in the auth service, instead of a get request, to populate the first and last name.

Just a thought, your call on which is better (and whether you're up for that much work :P).
